src/main/java/co/cask/PluginJSON.java

```diff
@@ -196,6 +216,63 @@ public void execute()  throws MojoExecutionException {
   }
 
   /**
+   * Inspects all the Icon files to generate properties to be a part of the related widget.
+   */
+  private JSONObject getIcon(List<String> fileNames, File[] files, File nameFile)
+    throws MojoExecutionException {
+
+    JSONObject object = new JSONObject();
+    String name = FileUtils.removeExtension(nameFile.getName());
+    if (fileNames.contains(name)) {
+      object.put("type", "inline");
+      Map<String, String> arguments = new TreeMap<String, String>();
+      File file = files[fileNames.indexOf(name)];
```

Comment:
using two lists like this makes things more error prone, a `Map<String, File>` seems like a much more natural way to represent the icons.